ui/src/components/table/QTable.sass

```
@@ -243,6 +243,8 @@
  */
 .q-table__bottom
   border-top: 1px solid $table-border-color
+  &.q-table__bottom--nodata
+    border-top: none
```

I haven't tested your solution or my suggestion yet, but I'm leaving it here for now
```suggestion
.q-table__bottom
  &:not(&--nodata)
    border-top: 1px solid $table-border-color
```

---

I tried your proposal, and changed my own code accordingly. It works and it is much cleaner than my initial thought. Thanks for pointing me out this alternative !